packages/grafana-ui/src/components/DataSourceSettings/Sigv4AuthSettings.tsx

```diff
@@ -0,0 +1,242 @@
+import React from 'react';
+import { HttpSettingsProps } from './types';
+import { SelectableValue } from '@grafana/data';
+import { Button, InlineFormLabel, Input } from '..';
+import Select from '../Forms/Legacy/Select/Select';
+
+export const Sigv4AuthSettings: React.FC<HttpSettingsProps> = props => {
+  const { dataSourceConfig } = props;
+
+  const authProviderOptions = [
+    { label: 'Access & secret key', value: 'keys' },
+    { label: 'Credentials file', value: 'credentials' },
+    { label: 'ARN', value: 'arn' },
+  ] as SelectableValue[];
+
+  const regions = [
```

Comment:
These are hardcoded for simplicity for now, but can be changed to dynamically fetch if necessary